src/hazelcore/OcamlExtraction.re

```diff
@@ -0,0 +1,374 @@
+module Typ = {
+  type ocaml_typ = string;
+
+  let rec extract = (~t: HTyp.t): ocaml_typ =>
+    switch (t) {
+    | Hole => "'a"
+    | Int => "int"
+    | Float => "float"
+    | Bool => "bool"
+    // add parenthesis is necessary
+    // (int -> int) -> int != int -> int -> int
+    | Arrow(t1, t2) =>
+      "(" ++ extract(~t=t1) ++ " -> " ++ extract(~t=t2) ++ ")"
+    | Sum(t1, t2) => "(" ++ extract(~t=t1) ++ " | " ++ extract(~t=t2) ++ ")"
```

Comment:
I don't think this is valid OCaml syntax for anonymous sums. You'll have to use polymorphic variants for this (you can read more about them online)

Reply:
I've read about the polymorphic variants. I'm a little bit worried that the variant needs a constructor, for example, if I encode the sum type as ` `Left of t | `Right of t `, then I need to explicitly use the constructor `Left or `Right for the expression. I'm wondering how the sum type expression is created in Hazel, can we directly declare something of a sum type?